### PR TITLE
Mini challenge 2

### PR DESCRIPTION
**Pull request by Danielalab**
Feedback: https://github.com/cvillanueva84/react-certification-2021/pull/68

### CODE REVIEW

File: src/components/Layout/Layout.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const { container } = render(<Layout />);
+    expect(container).toBeInTheDocument();
+    expect(container).toMatchSnapshot();
+  });
+
+  test('Should contains a "main" tag with "container" class', () => {
+    const { container } = render(<Layout />);
+    const mainElement = container.querySelector('main');
+    expect(mainElement).toBeInTheDocument();
+    expect(mainElement.classList.contains('container')).toBe(true);
```
Comment (moyomc) [inline src/components/Layout/Layout.spec.js]: There's also a `toContain` assertion:
```javascript
expect(mainElement.classList).toContain('container');
```

File: src/components/ToggleSwitch/ToggleSwitch.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  test('Should renders a ToggleStwitch component', () => {
+    expect(container).toMatchSnapshot();
+  });
+
+  test('Should contains a label with a switch role', () => {
+    const labelElement = screen.getByRole('switch');
+    expect(labelElement).toBeInTheDocument();
+  });
+  test('Should contains an input type checkbox', () => {
+    const inputElement = screen.getByTestId('input-checkbox');
```
Comment (moyomc) [inline src/components/ToggleSwitch/ToggleSwitch.spec.js]: Remember that querying byTestId should be our last option, ideally we want to query byRole (as you did in other cases). You can see the recommended priority guide here: https://testing-library.com/docs/queries/about/#priority

File: src/pages/Home/Home.page.spec.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import Home from './Home.page';
+
+const { render, screen } = require('@testing-library/react');
+
+jest.mock('../../components/VideosList/VideosList', () => {
```
Comment (moyomc) [inline src/pages/Home/Home.page.spec.js]: Interesting mock, for this test case it makes sense since that is the child component and will be dynamic :+1:

**Review by moyomc — APPROVED** [review]
Well done! These are good and meaningful test cases.
I left some minor comments.

Good on using getBy instead of queryBy 👍 

Please let me know if you have any comments or doubts

File: src/components/ToggleSwitch/ToggleSwitch.spec.js (hunk 2 above)
Comment (Danielalab) [reply src/components/ToggleSwitch/ToggleSwitch.spec.js]: Thanks @moyomc ! I've added a button role  to my input.

**Pull request merged**